ambry-tools/src/main/java/com.github.ambry/store/DumpDataHelper.java

```diff
@@ -92,6 +92,20 @@ static LogBlobRecordInfo readSingleRecordFromLog(RandomAccessFile randomAccessFi
                 + header.getUserMetadataRecordRelativeOffset() + " dataRelativeOffset "
                 + header.getBlobRecordRelativeOffset() + " crc " + header.getCrc();
         totalRecordSize += header.getMessageSize() + buffer.capacity();
+      } else if (version == MessageFormatRecord.Message_Header_Version_V3) {
+        ByteBuffer buffer = ByteBuffer.allocate(MessageFormatRecord.MessageHeader_Format_V3.getHeaderSize());
+        buffer.putShort(version);
+        randomAccessFile.read(buffer.array(), 2, buffer.capacity() - 2);
```

Comment:
Could you explain "- 2" here?

Reply:
2 is the number of bytes to store header version (it's a short). I copy this code from case above.